debugging/netcoreapp/dockerDebugConfigurationProvider.ts

```diff
@@ -0,0 +1,249 @@
+    windows?: DebugConfigurationBrowserBaseOptions;
+    osx?: DebugConfigurationBrowserBaseOptions;
+    linux?: DebugConfigurationBrowserBaseOptions;
+}
+
+interface DockerDebugConfiguration extends DebugConfiguration {
+    dockerOptions?: DockerDebugOptions;
+}
+
+export class DockerDebugConfigurationProvider implements DebugConfigurationProvider {
+    constructor(
+        private readonly debugSessionManager: DebugSessionManager,
+        private readonly dockerManager: DockerManager,
+        private readonly fsProvider: FileSystemProvider,
+        private readonly osProvider: OSProvider,
+        private readonly netCoreProjectProvider: NetCoreProjectProvider,
+        private readonly prerequisite: Prerequisite) {
+    }
+
+    public provideDebugConfigurations(folder: WorkspaceFolder | undefined, token?: CancellationToken): ProviderResult<DebugConfiguration[]> {
+        return [
+            {
+                name: 'Docker: Launch .NET Core (Preview)',
+                type: 'docker-netcoreapp',
+                request: 'launch',
+                preLaunchTask: 'build'
+            }
+        ];
+    }
+
+    public resolveDebugConfiguration(folder: WorkspaceFolder | undefined, debugConfiguration: DockerDebugConfiguration, token?: CancellationToken): ProviderResult<DebugConfiguration> {
+        return this.resolveDockerDebugConfiguration(folder, debugConfiguration);
+    }
+
+    private static resolveFolderPath(folderPath: string, folder: WorkspaceFolder): string {
+        return folderPath.replace(/\$\{workspaceFolder\}/g, folder.uri.fsPath);
+    }
+
+    private async resolveDockerDebugConfiguration(folder: WorkspaceFolder, debugConfiguration: DockerDebugConfiguration): Promise<DebugConfiguration> {
+        if (!folder) {
+            return undefined;
+        }
+
+        const prerequisiteSatisfied = await this.prerequisite.checkPrerequisite();
+
+        if (!prerequisiteSatisfied) {
+            return undefined;
+        }
+
+        const appFolder = this.inferAppFolder(folder, debugConfiguration);
+
+        const resolvedAppFolder = DockerDebugConfigurationProvider.resolveFolderPath(appFolder, folder);
+
+        const appProject = await this.inferAppProject(debugConfiguration, resolvedAppFolder);
+
+        const resolvedAppProject = DockerDebugConfigurationProvider.resolveFolderPath(appProject, folder);
+
+        const context = this.inferContext(folder, resolvedAppFolder, debugConfiguration);
+
+        const resolvedContext = DockerDebugConfigurationProvider.resolveFolderPath(context, folder);
+
+        let dockerfile = debugConfiguration.dockerOptions && debugConfiguration.dockerOptions.build && debugConfiguration.dockerOptions.build.dockerfile
+            ? DockerDebugConfigurationProvider.resolveFolderPath(debugConfiguration.dockerOptions.build.dockerfile, folder)
+            : path.join(appFolder, 'Dockerfile'); // TODO: Omit dockerfile argument if not specified or possibly infer from context.
```

Comment:
Are these TODOs required before shipping?  If not, I'd use "CONSIDER"

Reply:
Updated.